rpcs3/Emu/RSX/Program/ProgramStateCache.h

```diff
@@ -245,10 +246,15 @@ class program_state_cache
 				{
 					if (rsx_fp_invalidation_count != umax && prev_rsx_count == rsx_fp_invalidation_count)
 					{
-						return std::forward_as_tuple(prev_fp->second, true);
+						// Shader UCODE must be the same.
+						// Shader config changes are not tracked at the moment
+						// Compare manually
+						if (program_hash_util::fragment_program_compare::config_only(prev_fp->first, rsx_fp))
```

Comment:
Config changes are tracked by the fragment_state_dirty flag. Though I suppose there's no harm in double-checking here since the test is simple.